File: CRMSBI/DataModelFlutter/msgroupmenumdl.dart
 import "dart:convert";
 import "package:flutter/foundation.dart";
 import "../helper/tools.dart";

     
 class MsGroupMenuMdl {
     String kodeGroup;
     String kodeMenu;
     bool bView;
     bool bNew;
     bool bEdit;
     bool bDelete;
     bool bPrint;
     bool bProcess;
     bool bGenerate;
     bool bValidate;
     String dibuatOleh;
     DateTime waktuDibuat;
     String diubahOleh;
     DateTime waktuDiubah;
     
     MsGroupMenuMdl({
         @required this.kodeGroup,
         @required this.kodeMenu,
         @required this.bView,
         @required this.bNew,
         @required this.bEdit,
         @required this.bDelete,
         @required this.bPrint,
         @required this.bProcess,
         @required this.bGenerate,
         @required this.bValidate,
         @required this.dibuatOleh,
         @required this.waktuDibuat,
         @required this.diubahOleh,
         @required this.waktuDiubah,
     });
     
     MsGroupMenuMdl.fromJson(Map<String, dynamic> json) : 

         kodeGroup = json["KodeGroup"],
         kodeMenu = json["KodeMenu"],
         bView = json["bView"],
         bNew = json["bNew"],
         bEdit = json["bEdit"],
         bDelete = json["bDelete"],
         bPrint = json["bPrint"],
         bProcess = json["bProcess"],
         bGenerate = json["bGenerate"],
         bValidate = json["bValidate"],
         dibuatOleh = json["DibuatOleh"],
         waktuDibuat = deserializeDatetimeModel(json["WaktuDibuat"]),
         diubahOleh = json["DiubahOleh"],
         waktuDiubah = deserializeDatetimeModel(json["WaktuDiubah"]);     
     MsGroupMenuMdl.initModel() : 

         kodeGroup = null,
         kodeMenu = null,
         bView = null,
         bNew = null,
         bEdit = null,
         bDelete = null,
         bPrint = null,
         bProcess = null,
         bGenerate = null,
         bValidate = null,
         dibuatOleh = null,
         waktuDibuat = null,
         diubahOleh = null,
         waktuDiubah = null;     
     Map<String, dynamic> toJson() => { 
     

         "KodeGroup": kodeGroup,
         "KodeMenu": kodeMenu,
         "bView": bView,
         "bNew": bNew,
         "bEdit": bEdit,
         "bDelete": bDelete,
         "bPrint": bPrint,
         "bProcess": bProcess,
         "bGenerate": bGenerate,
         "bValidate": bValidate,
         "DibuatOleh": dibuatOleh,
         "WaktuDibuat": serializeDatetime(waktuDibuat),
         "DiubahOleh": diubahOleh,
         "WaktuDiubah": serializeDatetime(waktuDiubah)     };     
     static void deleteModel() {
         return saveString("MsGroupMenuMdl", "");
     }
     
     static Future<MsGroupMenuMdl> getData(String kodeGroup, String kodeMenu) async {
         try {
             var json = await postRequest("Model/GetMsGroupMenuMdl", {
             "kodeGroup": kodeGroup, "kodeMenu": kodeMenu
             });
             return MsGroupMenuMdl.fromJson(json);
          } catch (e) {
             return null;
          }
      }
      
      static MsGroupMenuMdl getModel() {
         try {
             String json = getString("MsGroupMenuMdl");
             Map<String, dynamic> oMap = jsonDecode(json);
             return MsGroupMenuMdl.fromJson(oMap);
         } catch (e) {
             return null;
         }
      }
      
      static void saveModel(MsGroupMenuMdl oModel) {
         return saveString("MsGroupMenuMdl", jsonEncode(oModel));
      }
      
      static void saveModelJson(Map<String, dynamic> oMap) {
         return saveString("MsGroupMenuMdl", jsonEncode(MsGroupMenuMdl.fromJson(oMap)));
      }
 }

File: CRMSBI/DataModel/MsGroupMenuMdl.cs
 using CRMSBI.Helper;
 using NPoco;
 using System;
 
 namespace CRMSBI.Data
 {

     [TableName("MsGroupMenu")]
     [PrimaryKey("KodeGroup, KodeMenu", AutoIncrement = false)]
     public partial class MsGroupMenuMdl
     {
     
         public string KodeGroup { get; set; }
     
         public string KodeMenu { get; set; }
     
         public bool bView { get; set; }
     
         public bool bNew { get; set; }
     
         public bool bEdit { get; set; }
     
         public bool bDelete { get; set; }
     
         public bool bPrint { get; set; }
     
         public bool bProcess { get; set; }
     
         public bool bGenerate { get; set; }
     
         public bool bValidate { get; set; }
     
         public string DibuatOleh { get; set; }
     
         public DateTime WaktuDibuat { get; set; }
     
         public string DiubahOleh { get; set; }
     
         public DateTime WaktuDiubah { get; set; }
     
         public static MsGroupMenuMdl GetData(string kodeGroup, string kodeMenu)
         {
             return MdlDtl.GetDataQuery<MsGroupMenuMdl>("SELECT * FROM MsGroupMenu WITH (NOLOCK,NOWAIT) WHERE KodeGroup = @0 AND KodeMenu = @1", kodeGroup, kodeMenu);
         }
         public static MsGroupMenuMdl GetDataTransaction(MdlDto oMdlDto, string kodeGroup, string kodeMenu)
         {
             return oMdlDto.GetData<MsGroupMenuMdl>(new {KodeGroup = kodeGroup, KodeMenu = kodeMenu});
         }
     }
 }

